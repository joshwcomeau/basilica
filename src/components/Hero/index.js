// eslint-disable-next-line no-unused-vars
import React, { Component, PropTypes } from 'react';
import classNames from 'classnames';

import Stick from '../Stick';
import Logo from '../Logo';
import InnerWrapper from '../InnerWrapper';
import './index.scss';
import defaultBackgroundImage from '../../images/placeholder-home-hero.jpeg';


const Hero = ({ children, backgroundImage }) => {
  const classes = classNames(['hero']);

  return (
    <div
      className={classes}
      style={{ backgroundImage: `url(${backgroundImage})` }}
    >
      <Stick className="sticky-logo">
        <InnerWrapper>
          <Logo />
        </InnerWrapper>
      </Stick>

      {children}
    </div>
  );
};

Hero.propTypes = {
  children: PropTypes.node,
  backgroundImage: PropTypes.string.isRequired,
};

Hero.defaultProps = {
  backgroundImage: defaultBackgroundImage,
};

export default Hero;
