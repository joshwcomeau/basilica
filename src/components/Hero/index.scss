@import '../../variables';

.hero {
  position: relative;
  z-index: 6;
  height: 75vh;
  background-size: cover;
  background-position: center center;

  .sticky-logo {
    position: absolute;
    left: 0;
    right: 0;
    bottom: 0;

    .inner-wrapper .inner-wrapper-contents {
      margin-left: 50px;
      margin-right: 50px;
    }
  }

  @media screen and (max-width: $mq-tablet) {

  }

  @media screen and (max-width: $mq-mobile) {

  }
}
