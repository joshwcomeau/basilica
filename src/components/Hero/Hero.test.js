/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import Hero from './index';

describe('Hero', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Hero />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
