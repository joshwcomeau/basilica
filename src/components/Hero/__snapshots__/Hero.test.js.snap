exports[`Hero renders correctly 1`] = `
<div
  className="hero"
  style={
    Object {
      "backgroundImage": "url(test-file-stub)"
    }
  }>
  <div
    className="sticky-logo is-not-stuck"
    style={undefined}>
    <h2
      className="logo">
      Basilica
    </h2>
  </div>
</div>
`;
